### PR TITLE
pruebas de unidad y integracion de SearchHost

### PR DESCRIPTION
**Pull request by adridean**

### CODE REVIEW

File: HostAbroadTests/src/test/integrationTests/ASSearchImpTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import com.business.User;
+import com.business.ASFactory.ASFactory;
+import com.business.ASSearch.*;
+
+public class ASSearchImpTest {
+	private ASSearchImp assearch;
+	
+	@Test
+	public void searchHostTest() { 
+		ArrayList<TUser> aux = new ArrayList<TUser>();
```
Comment (GasanNazer) [inline HostAbroadTests/src/test/integrationTests/ASSearchImpTest.java]: I do not know how this is working but the idea of JPA is to use Entity pattern- in your case User. Please keep in mind that  that and change your aux array. Also I suggest you to name variables better cuz it is hard to follow your code this way.

File: HostAbroadTests/src/test/integrationTests/CommandSearchHostTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public class CommandSearchHostTest {
+	private CommandSearchHost commandsh;
+
+	@Test
+	public void test() {
+		this.commandsh = new CommandSearchHost();
+		
+		Pair<Integer, Object> par, par_comando;
+		ArrayList<User> original = new ArrayList<User>();
+		ArrayList<TUser> lista_comando;
```
Comment (GasanNazer) [inline HostAbroadTests/src/test/integrationTests/CommandSearchHostTest.java]: Same as the previous file. TUser should be User. And name variables in English.

File: HostAbroadTests/src/test/integrationTests/CommandSearchHostTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		assertEquals(par.getLeft(), par_comando.getLeft());
+		for(int i = 0; i < original.size(); i++) {
+			assertEquals(lista_comando.get(i).getNickname(), original.get(i).getNickname());
+			assertEquals(lista_comando.get(i).getDescription(), original.get(i).getDescription());
+			assertEquals(lista_comando.get(i).getRating(), original.get(i).getRating(), 2);
+			assertEquals(lista_comando.get(i).getHost(), original.get(i).getHost());
+		}
+		
+	}
+
```
Comment (GasanNazer) [inline HostAbroadTests/src/test/integrationTests/CommandSearchHostTest.java:61]: This test is pretty clever. Nice one.

File: HostAbroadTests/src/test/integrationTests/FactoryCommandImpTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+public class FactoryCommandImpTest {
+	private FactoryCommandImp fcommand;
+	
+	@Test
+	public void test() {
+		this.fcommand = (FactoryCommandImp) FactoryCommand.getInstance();
+		Command csh = new CommandSearchHost();
+		Command csh_return = this.fcommand.parseCommand(Commands.CommandSearchHost);
+
```
Comment (GasanNazer) [inline HostAbroadTests/src/test/integrationTests/FactoryCommandImpTest.java:20]: This test is a little strange. But for now I do not hava a better solution so good job.

File: HostAbroadTests/src/test/unitTests/TUserTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+		TUser aux;
+		this.myUser = new TUser("Adri", 5, "estoy haciendo pruebas", false);
+		
+		aux = new TUser("Jose", 2, "este se modifica", true);
+		
+		aux.setNickname("Adri");
+		assertEquals(aux.getNickname(), this.myUser.getNickname());
+		
+		aux.setRating(5);
+		assertEquals(aux.getRating(), this.myUser.getRating(), 2);
```
Comment (GasanNazer) [inline HostAbroadTests/src/test/unitTests/TUserTest.java:25]: What is for the last paramether(2) in assertEquals(aux.getRating(), this.myUser.getRating(), 2);

**Comment by GasanNazer** [comment]
> _No description provided._

Overall I can say that this is good. I've added some suggestions and a question in the end. 
Good job! Keep doing the same.

**Review by SirTorresDev — COMMENTED** [review]
It's ok
